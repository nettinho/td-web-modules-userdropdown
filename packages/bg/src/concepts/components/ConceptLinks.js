import _ from "lodash/fp";
import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Segment, Grid, Table } from "semantic-ui-react";
import { FormattedMessage } from "react-intl";
import ConceptLinkRow from "./ConceptLinkRow";
import ConceptLinkActions from "./ConceptLinkActions";

export const ConceptLinks = ({ conceptLinks, conceptLinksLoading }) => (
  <Segment attached="bottom">
    <Grid>
      <Grid.Row>
        <Grid.Column>
          <ConceptLinkActions />
        </Grid.Column>
      </Grid.Row>
      <Grid.Row>
        <Grid.Column>
          {!_.isEmpty(conceptLinks) && (
            <Table celled striped compact selectable>
              <Table.Header>
                <Table.Row>
                  <Table.HeaderCell
                    content={<FormattedMessage id="conceptLink.ou" />}
                  />
                  <Table.HeaderCell
                    content={<FormattedMessage id="conceptLink.system" />}
                  />
                  <Table.HeaderCell
                    content={<FormattedMessage id="conceptLink.group" />}
                  />
                  <Table.HeaderCell
                    content={<FormattedMessage id="conceptLink.structure" />}
                  />
                  <Table.HeaderCell
                    content={<FormattedMessage id="conceptLink.field" />}
                  />
                  <Table.HeaderCell />
                </Table.Row>
              </Table.Header>
              <Table.Body>
                {conceptLinks.map((link, i) => (
                  <ConceptLinkRow key={i} {...link} />
                ))}
              </Table.Body>
            </Table>
          )}
          {_.isEmpty(conceptLinks) &&
            !conceptLinksLoading && (
              <FormattedMessage id="conceptLinks.empty" />
            )}
        </Grid.Column>
      </Grid.Row>
    </Grid>
  </Segment>
);

ConceptLinks.propTypes = {
  conceptLinksLoading: PropTypes.bool,
  conceptLinks: PropTypes.array
};

const mapStateToProps = ({ conceptLinks }) => ({
  conceptLinks
});

export default connect(mapStateToProps)(ConceptLinks);
