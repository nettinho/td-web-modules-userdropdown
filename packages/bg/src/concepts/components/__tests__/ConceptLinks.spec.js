import React from "react";
import { shallow } from "enzyme";
import { ConceptLinks } from "../ConceptLinks";

describe("<ConceptLinks />", () => {
  it("matches the latest snapshot (empty)", () => {
    const wrapper = shallow(<ConceptLinks />);
    expect(wrapper).toMatchSnapshot();
  });

  it("matches the latest snapshot (loading)", () => {
    const wrapper = shallow(<ConceptLinks conceptLinksLoading />);
    expect(wrapper).toMatchSnapshot();
  });

  it("matches the latest snapshot (non-empty)", () => {
    const field = {
      ou: "ou",
      system: "system",
      group: "group",
      structure: "structure",
      field: "id"
    };
    const conceptLinksLoading = false;
    const conceptLinks = [{ id: 1, field }, { id: 2, field }];
    const props = {
      conceptLinks,
      conceptLinksLoading
    };

    const wrapper = shallow(<ConceptLinks {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
