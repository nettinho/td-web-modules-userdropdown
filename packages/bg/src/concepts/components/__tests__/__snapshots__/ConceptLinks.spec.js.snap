// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConceptLinks /> matches the latest snapshot (empty) 1`] = `
<Segment
  attached="bottom"
>
  <Grid>
    <GridRow>
      <GridColumn>
        <Connect(ConceptLinkActions) />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <FormattedMessage
          id="conceptLinks.empty"
          values={Object {}}
        />
      </GridColumn>
    </GridRow>
  </Grid>
</Segment>
`;

exports[`<ConceptLinks /> matches the latest snapshot (loading) 1`] = `
<Segment
  attached="bottom"
>
  <Grid>
    <GridRow>
      <GridColumn>
        <Connect(ConceptLinkActions) />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn />
    </GridRow>
  </Grid>
</Segment>
`;

exports[`<ConceptLinks /> matches the latest snapshot (non-empty) 1`] = `
<Segment
  attached="bottom"
>
  <Grid>
    <GridRow>
      <GridColumn>
        <Connect(ConceptLinkActions) />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <Table
          as="table"
          celled={true}
          compact={true}
          selectable={true}
          striped={true}
        >
          <TableHeader
            as="thead"
          >
            <TableRow
              as="tr"
              cellAs="td"
            >
              <TableHeaderCell
                as="th"
                content={
                  <FormattedMessage
                    id="conceptLink.ou"
                    values={Object {}}
                  />
                }
              />
              <TableHeaderCell
                as="th"
                content={
                  <FormattedMessage
                    id="conceptLink.system"
                    values={Object {}}
                  />
                }
              />
              <TableHeaderCell
                as="th"
                content={
                  <FormattedMessage
                    id="conceptLink.group"
                    values={Object {}}
                  />
                }
              />
              <TableHeaderCell
                as="th"
                content={
                  <FormattedMessage
                    id="conceptLink.structure"
                    values={Object {}}
                  />
                }
              />
              <TableHeaderCell
                as="th"
                content={
                  <FormattedMessage
                    id="conceptLink.field"
                    values={Object {}}
                  />
                }
              />
              <TableHeaderCell
                as="th"
              />
            </TableRow>
          </TableHeader>
          <TableBody
            as="tbody"
          >
            <withRouter(Connect(ConceptLinkRow))
              field={
                Object {
                  "field": "id",
                  "group": "group",
                  "ou": "ou",
                  "structure": "structure",
                  "system": "system",
                }
              }
              id={1}
              key="0"
            />
            <withRouter(Connect(ConceptLinkRow))
              field={
                Object {
                  "field": "id",
                  "group": "group",
                  "ou": "ou",
                  "structure": "structure",
                  "system": "system",
                }
              }
              id={2}
              key="1"
            />
          </TableBody>
        </Table>
      </GridColumn>
    </GridRow>
  </Grid>
</Segment>
`;
